### PR TITLE
WIP pinning for kenny

### PR DESCRIPTION
**Pull request by yaahc**
cc @kennytm

### CODE REVIEW

**Review by kennytm — APPROVED** [review]
Rest LGTM

File: fakecore/src/any.rs [hunk 1]
```diff
@@ -48,6 +49,7 @@ impl<'out> Request<'out> {
             );
```
Comment (kennytm) [inline fakecore/src/any.rs:49]: this line 

```rust
            let this = unsafe { &mut *(self as *mut _ as *mut RequestBuf<Option<&'out T>>) };
```

probably needs to be changed to `self.get_unchecked_mut() as *mut _ as ...`.

**Comment by yaahc** [comment]
@kennytm did you see the other PR? Nika ended up doing the same work and I'm leaning towards merging her version over mine but I want to get your approval on it before I do.

**Comment by kennytm** [comment]
Just seen it. Replied on #2, feel free to close this one.

**Pull request closed without merging**